qutebrowser/mainwindow/statusbar/bar.py

```diff
@@ -215,6 +209,35 @@ def _on_config_changed(self, option):
             self.maybe_hide()
         elif option == 'statusbar.padding':
             self._set_hbox_padding()
+        elif option == 'statusbar.widgets':
+            self._draw_widgets()
+
+    def _draw_widgets(self):
+        """Draw statusbar widgets."""
+        # Start with widgets hidden and show them when needed
+        for widget in [self.url, self.percentage,
+                       self.backforward, self.tabindex]:
+            self._hbox.removeWidget(widget)
+            widget.hide()
+
+        # Read the list and set widgets accordingly
+        for segment in config.val.statusbar.widgets:
+            if segment == 'url':
+                self._hbox.addWidget(self.url)
+                self.url.show()
+            elif segment == 'scroll':
+                self._hbox.addWidget(self.percentage)
+                self.percentage.show()
+            elif segment == 'scroll_raw':
+                self._hbox.addWidget(self.percentage)
+                self.percentage.raw = True
+                self.percentage.show()
+            elif segment == 'history':
+                self._hbox.addWidget(self.backforward)
+                self.backforward.show()
```

Comment:
This shows the widget even if it's empty, causing a space to appear after changing the setting. This should probably only be shown if there's text, but then the `isHidden()` check in the widget will break. I guess you'll need some kind of `enabled` attribute you set from the outside instead.

Reply:
I tried this way but havent properly tested yet.